geocruncher/main.py

```diff
@@ -2,29 +2,56 @@
 
 import sys
 import numpy as np
-from .GeologicalModel3D import GeologicalModel
+#from .GeologicalModel3D import GeologicalModel
 
-if __name__ == '__main__':
-    main()
+def printStderr(msg):
+    sys.stderr.write(msg)
+    sys.stderr.flush()
 
 def main():
-    lines = '';
-    for line in sys.stdin:    
-        lines += line;
+    while True:
+        cmd = sys.stdin.readline().strip().split(" ")
+        if cmd[0] == 'SetProjectData':
+            xmlFile = cmd[1]
+            printStderr('Reading project XML data from ' + xmlFile + '\n')
+            # TODO
+        elif cmd[0] == "SetProjectDEM":
```

Comment:
The main idea here would be not to reload the DEM if it hasn't changed as it is kind of big. However, I am not sure if gmlib allows us to do that easily.